crates/nu-cli/src/completions/column_completions.rs

```diff
@@ -128,3 +120,51 @@ fn columns_to_suggestions(
         })
         .collect()
 }
+
+fn expr_allowed(expr: &Expr, working_set: &StateWorkingSet) -> bool {
+    log::debug!("expr={:?}", expr);
+
+    match expr {
+        Expr::String(..) => true,
+        Expr::FullCellPath(cell) => match &cell.head.expr {
+            Expr::Var(..) => true,
+            Expr::Record(v) => matches!(v.first(), Some((cols,_)) if cols.as_string().is_some()),
+            _ => false,
+        },
+        Expr::Call(call) => {
+            let command = working_set.get_decl(call.decl_id);
+            let category = command.signature().category;
+            let name = command.name();
+            let is_custom = command.is_custom_command();
+
+            // Only evaluate expressions without side effects
+            if is_custom {
+                false
+            } else if matches!(category, Category::Core) && name == "echo" {
+                true
+            } else {
+                category_allowed(category)
+            }
+        }
+        _ => false,
+    }
+}
+
+fn category_allowed(category: Category) -> bool {
+    matches!(
+        category,
+        Category::Bits
+            | Category::Bytes
+            | Category::Conversions
+            | Category::Date
+            | Category::Default
+            | Category::Filters
+            | Category::Formats
+            | Category::Generators
+            | Category::Hash
+            | Category::Math
+            | Category::Random
```

Comment:
`Category::Random` sounds like something were reads would be `volatile` or have a side effect in updating the RNG state.

I am not sure if the categories are all homogeneous enough to ensure that there is nothing with a side effect, yet.

Reply:
You're right, it's safe to filter out commands categorized as Category::Random.